### ISSUE TITLE
[Question] Blocking HTTP redirection (>300)

### ISSUE THREAD

**Issue by bodinsamuel** [microsoft/playwright#12527]
hello,

I'm trying to replace Puppeteer with Playwright and I'm currently stuck with this problem.
For Algolia Crawler, we use Puppeteer to render a page and abort on any navigation change for security and reporting reason (basically crawl each redirection separately).

I'm trying to block HTTP 300 however I can not find any way to do this.
In Puppeteer you can do something [like](https://stackoverflow.com/questions/46886832/how-to-stop-puppeteer-follow-redirects):
```js
page.on('request', request => {
  if (request.isNavigationRequest() && request.redirectChain().length !== 0) {
    request.abort();
  } else {
    request.continue();
  }
});
```

However in Playwright `on('request')` is read-only, so I'm using `route('**/*')`, but my route never receive the event for 301.

Basically:
```js
goto('http://localhost/301');
=> enter route
=> enter on('request')
=> enter on('response')
301 kicks in
=> enter on('request')
=> enter on('response')
```

It might be a bug or completely normal but if it's normal can you advice on how to block the redirection?
Best Regards,

**Comment by yury-s** [comment]
Playwright doesn't support interception of redirected requests, you can only abort/override original request sent from the page.

**Comment by bodinsamuel** [comment]
Thanks for confirming.
Is there any reasons for it to be that way?

I assumed, if it was possible in puppeteer it should be somewhat doable in playwright?
Would you recommend anything to achieve it? Maybe with `CDPSession`?

For most users it is probably a non-issue since they render their own site so I understand it's not top prio, 
however redirection can have security implication when you allow users to render any webpages or inject JS. (301, or http-equiv, or js redirect).

Thanks,

**Comment by yury-s** [comment]
> Thanks for confirming. Is there any reasons for it to be that way?
> 
There are technical limitation that prevented us from doing it in other browsers, e.g. in WebKit on macOS the redirects are handled in the OS without hooks that would allow to intercept intermediate responses. We'd like to have consistent API across browsers.

> Would you recommend anything to achieve it? Maybe with `CDPSession`?

You can always use CDPSession to send raw CDP commands, it will allow you to intercept redirects as well.

Closing the issues as there are no plans to implement this.

**Comment by bodinsamuel** [comment]
Fair. This could deserves a little banner in the documentation imo

**Comment by EMinasian** [comment]
> Fair. This could deserves a little banner in the documentation imo

Hey, not sure if you have come up with this: https://playwright.dev/docs/api/class-page#page-route
